src/analysis/souffle/authorization_logic.dl

```diff
@@ -86,4 +100,11 @@ ownsTag(owner, tag) :- says_ownsTag(_, owner, tag).
 isPrincipal(owner) :- ownsTag(owner, _).
 isPrincipal(owner) :- ownsAccessPath(owner, _).
 
+isPrincipal(speaker), isAccessPath(ap), isPrincipal(owner), isTag(tag) :-
```

Comment:
I didn't know you could write rules with multiple heads (or I forgot). Nice, that's easier to read.

Reply:
Isn't it nice? Especially when population should be correlated, as in this case.